src/foundry/client-esm/data/calendar.d.mts

```diff
@@ -201,12 +202,125 @@ declare namespace CalendarData {
     // TODO: Replace with CreateData and convert to interface
     type CreateData = fields.SchemaField.AssignmentType<Schema>;
   }
+
+  /**
+   * A decomposition of the integer world time in seconds into component parts.
+   * Each component expresses the number of that temporal unit since the time=0 epoch.
+   */
+  interface TimeComponents {
+    /** The number of years completed since zero */
+    year: number;
+
+    /** The number of days completed within the year */
+    day: number;
+
+    /** The number of hours completed within the year */
+    hour: number;
+
+    /** The number of minutes completed within the hour */
+    minute: number;
+
+    /** The number of seconds completed within the minute */
+    second: number;
+
+    /** The month, an index of the months.values array */
+    month: number;
+
+    /** The day of the month, starting from zero */
+    dayOfMonth: number;
+
+    /** The weekday, an index of the days.values array */
+    dayOfWeek: number;
+
+    /** The season, an index of the seasons.values array */
+    season: number;
+
+    /** Is it a leap year? */
+    leapYear: boolean;
+  }
+
+  // TODO: Reference CONFIG
+  type Formatters = "formatTimestamp" | "formatAgo";
+
+  /**
+   * @returns The returned string format
+   */
+  type TimeFormatter<Options> = (
+    /** The configured calendar */
+    calendar: CalendarData,
+    /** Time components to format */
+    components: TimeComponents,
+    /** Additional formatting options */
+    options: Options,
+  ) => string;
+
+  interface FormatAgoOptions {
+    short?: boolean | undefined;
+
+    maxTerms?: number | undefined | null;
+  }
 }
 
 /**
  * Game Time Calendar configuration data model.
  */
-declare class CalendarData extends DataModel<CalendarData.Schema> {}
+declare class CalendarData extends DataModel<CalendarData.Schema> {
+  /**
+   * Expand a world time integer into an object containing the relevant time components.
+   * @param components - An amount of time expressed as components
+   * @returns The cumulative time in seconds
+   */
+  componentsToTime(components: NullishProps<CalendarData.TimeComponents>): number;
```

Comment:
Can you extract this out to an interface? Maybe let's call it `InputTimeComponent`?

My goal is to avoid as many transformations directly in a function signature as possible, though this does pose some challenges.

In this case it's reused enough that I think it works.